ompi/mca/op/rocm/op_rocm_component.c

```diff
@@ -0,0 +1,219 @@
+};
+
+/*
+ * Component open
+ */
+static int rocm_component_open(void)
+{
+    /* We checked the flags during register, so if they are set to
+     * zero either the architecture is not suitable or the user disabled
+     * AVX support.
+     *
+     * A first level check to see what level of AVX is available on the
+     * hardware.
+     *
+     * Note that if this function returns non-OMPI_SUCCESS, then this
+     * component won't even be shown in ompi_info output (which is
+     * probably not what you want).
+     */
+    return OMPI_SUCCESS;
+}
+
+/*
+ * Component close
+ */
+static int rocm_component_close(void)
+{
+    if (mca_op_rocm_component.ro_num_devices > 0) {
+        //hipStreamDestroy(mca_op_rocm_component.ro_stream);
+        free(mca_op_rocm_component.ro_max_threads_per_block);
+        mca_op_rocm_component.ro_max_threads_per_block = NULL;
+        free(mca_op_rocm_component.ro_max_blocks);
+        mca_op_rocm_component.ro_max_blocks = NULL;
+        free(mca_op_rocm_component.ro_devices);
+        mca_op_rocm_component.ro_devices = NULL;
+        mca_op_rocm_component.ro_num_devices = 0;
+    }
+
+    return OMPI_SUCCESS;
+}
+
+/*
+ * Register MCA params.
+ */
+static int
+rocm_component_register(void)
+{
+    /* TODO: add mca paramters */
+
+    mca_base_var_enum_flag_t *new_enum_flag = NULL;
+    (void) mca_base_component_var_register(&mca_op_rocm_component.super.opc_version,
+                                           "max_num_blocks",
+                                           "Maximum number of thread blocks in kernels (-1: device limit)",
+                                           MCA_BASE_VAR_TYPE_INT,
+                                           &(new_enum_flag->super), 0, 0,
+                                           OPAL_INFO_LVL_4,
+                                           MCA_BASE_VAR_SCOPE_LOCAL,
+                                           &mca_op_rocm_component.ro_max_num_blocks);
+
+    (void) mca_base_component_var_register(&mca_op_rocm_component.super.opc_version,
+                                           "max_num_threads",
+                                           "Maximum number of threads per block in kernels (-1: device limit)",
+                                           MCA_BASE_VAR_TYPE_INT,
+                                           &(new_enum_flag->super), 0, 0,
+                                           OPAL_INFO_LVL_4,
+                                           MCA_BASE_VAR_SCOPE_LOCAL,
+                                           &mca_op_rocm_component.ro_max_num_threads);
+
+    return OMPI_SUCCESS;
+}
+
+
+/*
+ * Query whether this component wants to be used in this process.
+ */
+static int
+rocm_component_init_query(bool enable_progress_threads,
+                         bool enable_mpi_thread_multiple)
+{
+    int num_devices;
+    int rc;
+    int prio_lo, prio_hi;
+    //memset(&mca_op_rocm_component, 0, sizeof(mca_op_rocm_component));
+    hipInit(0);
```

Comment:
hipInit(0) should not be necessary, the function is not actually required.

Reply:
Fixed.